extensions/amp-vimeo/0.1/amp-vimeo.js

```diff
@@ -0,0 +1,78 @@
+/**
+ * Copyright 2015 The AMP HTML Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {isLayoutSizeDefined} from '../../../src/layout';
+import {loadPromise} from '../../../src/event-helper';
+
+class AmpVimeo extends AMP.BaseElement {
+
+  /** @override */
+  preconnectCallback(onLayout) {
+    this.preconnect.url('https://player.vimeo.com', onLayout);
+    // Host that Vimeo uses to serve poster frames needed by player.
+    this.preconnect.url('https://i.vimeocdn.com', onLayout);
+    // Host that Vimeo uses to serve JS, CSS and other assets needed.
+    this.preconnect.url('https://f.vimeocdn.com', onLayout);
+  }
+
+  /** @override */
+  isLayoutSupported(layout) {
+    return isLayoutSizeDefined(layout);
+  }
+
+  /** @override */
+  layoutCallback() {
+    const width = this.element.getAttribute('width');
+    const height = this.element.getAttribute('height');
+    // The video-id is supported only for backward compatibility.
+    const videoid = AMP.assert(
+        (this.element.getAttribute('data-videoid') ||
+        this.element.getAttribute('video-id')),
```

Comment:
No support for `video-id`. We changed our mind about that (YT was like the 1st component we ever created and it had that).